instituto-brasileiro-de-informacao-em-ciencia-e-tecnologia-abnt.csl

```diff
@@ -291,104 +393,280 @@
+        <name-part name="family"/>
+      </name>
+      <et-al font-style="italic"/>
+    </names>
   </macro>
-  <macro name="number">
-    <group>
-      <text variable="number" prefix=", n. "/>
-    </group>
+  <macro name="publisher">
+    <choose>
+      <if type="article" match="any">
+        <text variable="publisher" font-weight="bold"/>
+      </if>
+      <else-if type="bill broadcast chapter entry-dictionary entry-encyclopedia map motion_picture paper-conference report software song" match="any">
+        <choose>
+          <if variable="publisher-place publisher" match="any">
+            <group delimiter=": ">
+              <choose>
+                <if variable="publisher-place">
+                  <text variable="publisher-place"/>
+                </if>
+                <else>
+                  <group>
+                    <text value="["/>
+                    <text term="no-place" form="short" text-case="capitalize-first" font-style="italic"/>
+                    <text value="]"/>
+                  </group>
+                </else>
+              </choose>
+              <choose>
+                <if variable="publisher">
+                  <text variable="publisher"/>
+                </if>
+                <else>
+                  <group>
+                    <text value="["/>
+                    <text term="no-publisher" form="short" text-case="capitalize-first" font-style="italic"/>
+                    <text value="]"/>
+                  </group>
+                </else>
+              </choose>
+            </group>
+          </if>
+        </choose>
+      </else-if>
+      <else>
+        <text value="["/>
+        <group delimiter=": ">
+          <text term="no-place" form="short" text-case="capitalize-first" font-style="italic"/>
+          <text term="no-publisher" form="short" font-style="italic"/>
+        </group>
+        <text value="]"/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-issued">
+    <choose>
+      <if variable="publisher-place publisher" match="any">
+        <group delimiter=": ">
+          <choose>
+            <if variable="publisher-place">
+              <text variable="publisher-place"/>
+            </if>
+            <else>
+              <group>
+                <text value="["/>
+                <text term="no-place" form="short" text-case="capitalize-first" font-style="italic"/>
+                <text value="]"/>
+              </group>
+            </else>
+          </choose>
+          <choose>
+            <if variable="publisher">
+              <text variable="publisher"/>
+            </if>
+            <else>
+              <group>
+                <text value="["/>
+                <text term="no-publisher" form="short" text-case="capitalize-first" font-style="italic"/>
+                <text value="]"/>
+              </group>
+            </else>
+          </choose>
+        </group>
+      </if>
+      <else>
+        <text value="["/>
```

Comment:
please try to avoid value throughout your style. It's only necessary in very few cases.